src/lib/errors/errorHandler.js

```diff
@@ -1,12 +1,22 @@
 import { NextResponse } from 'next/server';
 import { ZodError } from 'zod';
 import { BASE_ERROR_MESSAGES } from '../consts';
+import { NotAuthorizedException } from '@/lib/errors/NotAuthorizedException';
 
 export function withErrorHandler(fn) {
   return async function handle(request, ...args) {
     try {
       return await fn(request, ...args);
     } catch (error) {
+      if (error instanceof NotAuthorizedException) {
```

Comment:
🟢 Are there any particular reasons to handle Not Authorized separately here?

Reply:
Wouldn't it be nice to catch all possible errors in one place?

Reply:
You may throw new NotAuthorizedException() from the place where it possibly can occure and handler will handle it automatically(look lines 30-39)

Reply:
NotAuthorziedExcpetion error object already has coressponding status and message properties

Reply:
Oh, I see what you are saying. Then yes, it shouldn't be there.

Reply:
Removed that block.